### PR TITLE
really fix print statement button flickering

### PR DESCRIPTION
**Pull request by souporserious**
This attempts to fix print statement button flickering by only rendering one element to the DOM and then translating it based on the current hovered line. I went down a small rabbit hole to get max performance by using refs and bypassing React, but it was making things more complicated so sticking with React state for now.

## Before

https://user-images.githubusercontent.com/2762082/192340437-74d08ca1-fab9-44de-ae8b-b46b948dc844.mov

## After

https://user-images.githubusercontent.com/2762082/192340471-28ef5695-19a1-42c8-b5ca-274d913de446.mov

## Future Considerations

The tooltip still sticks out as a little erratic when hovering lines. I think we can clean this up with a small overhaul to merge the print statement button, hit counts gutter, and tooltip into one state.

### CODE REVIEW

File: src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx
Comment (jasonLaster) [reply src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx:105]: we might want to move the clear/and set into one debounced update call

**Comment by jasonLaster** [comment]
@souporserious happy to review this once the types pass and i can view the preview branch. Would probably benefit from getting a code review from brian / mark.

File: src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx
Comment (souporserious) [reply src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx]: Moved the debounce back for now. Since this PR is more focused around `ToggleWidgetButton` I think we can tackle this in a follow-up.

**Comment by souporserious** [comment]
@jasonLaster ready for a look [here](https://devtools-oirief9mx-recordreplay.vercel.app/) 😎

File: src/devtools/client/debugger/src/components/Editor/YoYo.tsx
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/Editor/YoYo.tsx:1]: testing me? 😉

File: src/devtools/client/debugger/src/utils/editor/create-editor.ts
Comment (jasonLaster) [inline src/devtools/client/debugger/src/utils/editor/create-editor.ts:45]: oh nice

File: src/devtools/client/debugger/src/components/Editor/ToggleWidgetButton.tsx
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/Editor/ToggleWidgetButton.tsx:174]: awesome

File: src/devtools/client/debugger/src/components/Editor/ToggleWidgetButton.tsx
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/Editor/ToggleWidgetButton.tsx:161]: very cool

**Review by jasonLaster — COMMENTED** [review]
Looking really good

File: src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx:131]: we should cancel

**Review by jasonLaster — APPROVED** [review]

File: src/devtools/client/debugger/src/components/Editor/ToggleWidgetButton.tsx
Comment (souporserious) [inline src/devtools/client/debugger/src/components/Editor/ToggleWidgetButton.tsx:154]: This is no longer needed from what I can tell since we're switching the editor to `readOnly: "nocursor"` down below.

File: src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx
Comment (souporserious) [reply src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx]: Fixed!

File: src/devtools/client/debugger/src/components/Editor/YoYo.tsx
Comment (souporserious) [reply src/devtools/client/debugger/src/components/Editor/YoYo.tsx]: I think you were testing me 😂

**Comment by bvaughn** [comment]
A ton of CI stuff failed on this PR. Is that known/expected?

**Comment by bvaughn** [comment]
Travis and I chatted out of band about this and decided it was easiest to revert this temporarily and re-land it once #7790 lands. That PR changes so many things about the Toggle Widget Button component that it's hard to apply these changes on top of that. I can own that change if you all are too busy or I can pair etc.

**Pull request merged**
